eng/scripts/packagevalidation/PrePublish-Validation.ps1

```diff
@@ -0,0 +1,97 @@
+[CmdletBinding()]
+Param (
+  [ValidateNotNullOrEmpty()]
+  [string] $PackagePath,
+  [string] $ArtifactName,
+  [string] $WorkingDirectory
+)
+
+. "${PSScriptRoot}\..\common\scripts\logging.ps1"
+. "${PSScriptRoot}\..\common\scripts\SemVer.ps1"
+
+$WinVersion = (Get-ChildItem -Path "C:\Program Files (x86)\Windows Kits\10\bin\10.0*" | Sort-Object -Descending)[0]
+$SignTool = "C:\Program Files (x86)\Windows Kits\10\bin\$($WinVersion.Name)\x64\signtool.exe"
+
+New-Item -Path $WorkingDirectory -Name "Validation" -ItemType "directory"
+$ValidationDirectory = "$WorkingDirectory\Validation"
+
+Copy-Item -Path $PackagePath -Destination "$ValidationDirectory\package.zip"
+Expand-Archive -LiteralPath "$ValidationDirectory\package.zip" "$ValidationDirectory\extracted"
+
+$PackageName = Split-Path -Path $PackagePath -Leaf
+$PackageDlls = Get-ChildItem -Path "$ValidationDirectory\extracted\**\*.dll" -Recurse
+$NuspecFile = Get-ChildItem -Path "$ValidationDirectory\extracted\**\*.nuspec" -Recurse
+
+$NuspecContent = new-object xml
+$NuspecContent.Load($NuspecFile[0].FullName)
+$PackageId = $NuspecContent.package.metadata.id
+$PackageVersion = $NuspecContent.package.metadata.version
+$ProjectUrl = $NuspecContent.package.metadata.projectUrl
+$Description = $NuspecContent.package.metadata.description
+
+$IsValidPackage = $true
+
+LogDebug "Validating that Binaries and Packages are Signed..."
+foreach ($file in $PackageDlls) {
+    &$SignTool verify /pa $file.FullName
```

Comment:
Does signtool work on the package itself? If so we should run it against that as well.

Reply:
The sign tool does not work on the .nuget package.

Reply:
Ok we will need to see how we can validate that the package itself is signed then.